packages/medusa/src/services/customer.ts

```diff
@@ -476,21 +477,25 @@ class CustomerService extends BaseService {
     })
   }
 
-  async addAddress(customerId, address) {
-    return this.atomicPhase_(async (manager) => {
+  async addAddress(
+    customerId: string,
+    address: AddressCreatePayload
+  ): Promise<Customer | Address> {
+    return await this.atomicPhase_(async (manager) => {
       const addressRepository = manager.getCustomRepository(
         this.addressRepository_
       )
 
-      address.country_code = address.country_code.toLowerCase()
+      address.country_code = address.country_code.toLowerCase() ?? null
```

Comment:
todo: shoul we add some nullish coalescing? since we are looking for null or undefined to set to null, it means that country_code could be null or undefined and therefore the lowerCase would not work

```ts
address.country_code = address?.country_code?.toLowerCase() ?? null
```

Reply:
I have removed the null coalescing since the country_code is required when adding an address